pkg/lib/cockpit-components-table.jsx

```diff
@@ -75,9 +75,37 @@ export const ListingTable = ({
     let rows = tableRows;
 
     const [expanded, setExpanded] = useState({});
+    const [newItems, setNewItems] = useState([]);
+    const [currentRowsKeys, setCurrentRowsKeys] = useState([]);
     const [activeSortIndex, setActiveSortIndex] = useState(sortBy ? sortBy.index : 0);
     const [activeSortDirection, setActiveSortDirection] = useState(sortBy ? sortBy.direction : SortByDirection.asc);
 
+    useEffect(() => {
+        const getRowKeys = rows => {
+            const keys = [];
+
+            rows.forEach(row => {
+                if (row.props && row.props.key)
+                    keys.push(row.props.key);
+            });
+
+            return keys;
+        };
+
+        const current_keys = getRowKeys(rows);
+        if (JSON.stringify(current_keys) === JSON.stringify(currentRowsKeys))
+            return;
+
+        // Don't highlight all when the list gets loaded
+        if (currentRowsKeys.length !== 0) {
+            const new_keys = current_keys.filter(key => currentRowsKeys.indexOf(key) === -1);
+            if (new_keys.length)
+                setNewItems([...newItems, ...new_keys]);
```

Comment:
This only ever gets added to, instead of just showing the actually new keys once rows changes. I know the animation will eventually fade out, but over time it would mean that all rows are "new"? Why not `setNewItems(new_keys)` here?

Reply:
Mostly I am worried that when two items would be added in quick succession that it would remove this new class from the freshly added item. I haven't tested what happens when you remove class mid animation but leaving the class there has no negative impact.